### PR TITLE
Improved README file

### LINKED ISSUE #56: Readme file needs enhancement
## What's the problem or feature?

The readme file has some grammar and sentence structure issues. It also needs steps on how to use the project and setup it up for yourself.

## For the solution we want to

Fix some grammar issues and sentence structure problems that we have right now in our README.md file.

[And this is the result.](https://github.com/issavejdani/NMock/tree/doc-fix)

### PR DESCRIPTION
**Pull request by issavejdani**
Updated and improved the readme file grammatically and structurally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,38 +1,70 @@
+- [:collision: What is NMock?](#collision-what-is-nmock)
```
Comment (AbolfaZlRezaEe) [inline README.md:3]: Actually, the purpose of designing the NMock banner was meant to be on the top of the README.md file. It's good to see an overview of the application at the start and then start to read the documentation

File: README.md [hunk 2]
```diff
@@ -1,38 +1,70 @@
...
+**NMock** is a tool that helps you generate fake locations and trips to test the map feature in your product. It enables you to share them with others as well. We've built this application with the help of the following products and services:
+- **[Google Map SDK](https://developers.google.com/maps/documentation)**
+-  **[Kotlin Language](https://kotlinlang.org/)** 
+- **[Kotlin Coroutines](https://kotlinlang.org/docs/coroutines-overview.html)** 
+- **[Dagger-Hilt Dependency Injection](https://dagger.dev/hilt/)**. 
+
 ![NMock Banner](https://user-images.githubusercontent.com/73066290/198220049-159f1118-b181-415d-bc61-c9a0d7b1bf87.png)
 
-> Maybe you think about how map android developers test their features and functionalities? should every person that wanna test the map applications have a car and go to the streets to test? actually, the answer is No! in Android, we have some applications that we call them "Mock", and these applications help us to generate fake locations or fake trips to test our Map application and for other purposes.
+This document is a guide on how this application works.
```
Comment (AbolfaZlRezaEe) [inline README.md]: I think now we know that this README wanna explain the application in detail. so this does not necessarily need to write.

File: README.md [hunk 3]
```diff
@@ -1,38 +1,70 @@
...
 
-in ViewModels, we have a different approach for giving access to the Views. every ViewModel can have two outputs and Views can use those for managing their views and actions. at the first, we have a **Stateflow** that represents the state for the view. this flow contains all of the information that the view needs to show(like MVI approach).
 
-and the second output that works with **Sharedflow** sends some actions that can have a message or not. Views parse these actions and then, do an action in view like showing an error and so on... for example:
+In ViewModels, we have two different approaches to accessing the views. Each ViewModel has two outputs that can be used by Views to manage their views and actions:
+- The first one is a **Stateflow** which represents the state of the view. This flow contains all the information that the view needs to display (MVI approach)
+- The second one works with a **Sharedflow** and sends some actions that can either contain a message or be empty. View, then parse these actions and perform several actions, like displaying an error, for example.
```
Comment (AbolfaZlRezaEe) [inline README.md]: I think we use "action" in this sentence a lot. maybe it's good to write it like this:

`...View, then parse these actions and perform several works...`

File: README.md [hunk 4]
```diff
@@ -69,17 +101,17 @@ private fun initObservers() {
...
 ## :arrow_left::arrow_right: Mock Import/Export file structure
 
-In some situations, you need to share your trip with your friends or your team workers. we have a feature that gives access you, to share your mock and send it everywhere you want. but what's the structure of this export file?
+Sometimes you need to share your trip data with your friends or colleagues. We have implemented a feature allowing you to share your mocked data in JSON format easily.
 
-### Exporting Structure
+### Export
 
-as you maybe know, we save your mock trip and the state of that. for exporting the mock trip, we use **JSON** as a format of the file. the structure of this file is like this:
+We store your mock trip data and its state in **JSON** format. We use the same format to export this data. The following JSON is how we structure this data to export:
```
Comment (AbolfaZlRezaEe) [inline README.md]: The first sentence is wrong programmatically. We store mock trips in the database.

File: README.md [hunk 5]
```diff
@@ -122,32 +154,37 @@ as you maybe know, we save your mock trip and the state of that. for exporting t
...
+- fileCreatedAt
+- fileOwner
+- versionCode
 
-as you can see in exporting section, you can share a `.json` file to your friends everywhere you want. for importing it, we can **Only** receive this `.json` file in the import section of the application(for the future, We have a plan for importing mocks in many ways!).
+We separated the imported data so we could manage them easily. These are the reasons that give us the ability to do so:
 
-We save this `Json` file information like normal mocks but in a different table in the application database. We also save some extra information like: `fileCreatedAt`, `fileOwner` and `versionCode` in imported mock table. the reason that We split these mocks from normal mocks that you create in the application is, maybe the information of imported mocks can be less than normal mocks in the future. also, We have a plan for a new UI to show imported mocks in a different place in the application. so we create a new place for saving these mocks to managing them very easily in the future.
+- We may have less imported data in the future.
+- We have a plan to implement a separate UI to display imported data.
```
Comment (AbolfaZlRezaEe) [inline README.md]: Let me replace and improve this section with the following sentences:

- Maybe We have a different structure for the imported mocks in the future
- In the future, We wanna implement a different flow for showing imported mocks in UI and Logic too.

**Review by AbolfaZlRezaEe — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (issavejdani) [reply README.md]:
```suggestion
```

File: README.md (hunk 3 above)
Comment (issavejdani) [reply README.md]:
```suggestion
- The second one works with a **Sharedflow** and sends some actions that can either contain a message or be empty. View, then parse these actions and perform several works, like displaying an error, for example. 
```

File: README.md (hunk 4 above)
Comment (issavejdani) [reply README.md]:
```suggestion
We store your mock trip data and its state in the database. We use the same format to export this data. The following JSON is how we structure this data to export:
```

File: README.md (hunk 5 above)
Comment (issavejdani) [reply README.md]:
```suggestion
- Maybe We have a different structure for the imported mocks in the future
- In the future, We wanna implement a different flow for showing imported mocks in UI and Logic too.
```

File: README.md [hunk 6]
```diff
@@ -1,38 +1,70 @@
...
+
+There are applications in Android called Mock applications. They help us generate fake locations and trips to test map features. 
+
+**NMock** is a tool that helps you generate fake locations and trips to test the map feature in your product. It enables you to share them with others as well. We've built this application with the help of the following products and services:
+- **[Google Map SDK](https://developers.google.com/maps/documentation)**
+-  **[Kotlin Language](https://kotlinlang.org/)** 
+- **[Kotlin Coroutines](https://kotlinlang.org/docs/coroutines-overview.html)** 
+- **[Dagger-Hilt Dependency Injection](https://dagger.dev/hilt/)**. 
+
 ![NMock Banner](https://user-images.githubusercontent.com/73066290/198220049-159f1118-b181-415d-bc61-c9a0d7b1bf87.png)
```
Comment (issavejdani) [inline README.md]:
```suggestion
```

File: README.md (hunk 1 above)
Comment (issavejdani) [reply README.md:3]:
```suggestion
![NMock Banner](https://user-images.githubusercontent.com/73066290/198220049-159f1118-b181-415d-bc61-c9a0d7b1bf87.png)

- [:collision: What is NMock?](#collision-what-is-nmock)
```

**Review by AbolfaZlRezaEe — APPROVED** [review]

**Pull request merged**